### ISSUE TITLE
Can we have an option to not report IDE0009 (this. prefix) when a member starts with underscore?

### ISSUE THREAD

**Issue by lostmsu** [dotnet/roslyn#30130]
**Version Used**: 15.8

**Steps to Reproduce**:
```csharp
class A {
  int _x;
  A() { _x = 10; }
}
```

**Expected Behavior**:
IDE0009 not reported based on a setting (probably better have it not report by default), that allows skipping `this.` if member starts with '_'

**Actual Behavior**:
IDE0009 is reported

**Rationale**
There're two common guidelines for accessing instance members: make it clear with `this.` or prefix their names with '_'. The first is more common, but the second is also used in some projects, and probably should be kept in them for consistency.

It will be useful to have on projects like https://github.com/Microsoft/PTVS

**Comment by CyrusNajmabadi** [comment]
@lostmsu Why not turn off the rule that says:

![image](https://user-images.githubusercontent.com/4564579/46036918-9ec61880-c0bb-11e8-908e-efcb52426fbe.png)

?

If you name your fields with `_`, then there's little need for the 'this' rule for them since the naming makes it clear that it's a field.

**Comment by lostmsu** [comment]
@CyrusNajmabadi it makes sense for projects, that have mixed style to mandate `this.` for any field, whose name does not start with '_'

**Comment by sharwell** [comment]
> The first is more common, but the second is also used in some projects, and probably should be kept in them for consistency.

You can have multiple .editorconfig files in your solution by placing them in the project directory. Roslyn uses this to handle some style differences between the compiler projects and the rest.

**Comment by lostmsu** [comment]
@sharwell that assumes, that older projects must have all old code, and all new code to be isolated per-file, which is impractical for large codebases.

**Comment by CyrusNajmabadi** [comment]
@lostmsu Would you be willing to come over to gitter to chat?  I'd like to understand your use case a bit more, and it might be easier to just talk about it instead of having to asynchronously go back and forth :)
